src/crab_utils/bignums.hpp

```diff
@@ -138,6 +167,38 @@ class z_number final {
         }
     }
 
+    // Allow casting to uint32_t or uint64_t as needed for finite width operations.
+    [[nodiscard]] z_number cast_to_unsigned_finite_width(int finite_width) const {
+        switch (finite_width) {
+        case 0: return *this; // No finite width.
+        case 32: return cast_to_uint32();
+        case 64: return cast_to_uint64();
+        default: CRAB_ERROR("invalid finite width");
```

Comment:
That's my understand (within the crab directory), yes.  CRAB_ERROR outputs an error and throws an exception.  And this case here should never happen since the callers are part of prevail, which only pass 0, 32, or 64.